src/assets.ts

```diff
@@ -541,28 +543,29 @@ export async function addTasksJsonIfNecessary(generator: AssetGenerator, operati
             return resolve();
         }
 
+        
         const tasksJson = generator.createTasksConfiguration();
 
-        // NOTE: We only want to do this when we are supposed to update the task configuration. Otherwise,
-        // in the case of the 'generateAssets' command, even though we already deleted the tasks.json file
-        // this will still return the old tasks.json content
-        if (operations.updateTasksJson) {
-            const tasksConfigs = vscode.workspace.getConfiguration('tasks');
-            let existingTaskConfigs = tasksConfigs.get<Array<tasks.TaskDescription>>('tasks');
+        if (!fs.pathExistsSync(generator.tasksJsonPath)) {
+            // when tasks.json does not exist create it and write all the content
+            const tasksJsonText = JSON.stringify(tasksJson, null, '    ');
+            fs.writeFile(generator.tasksJsonPath, tasksJsonText, err => {
+                if (err) {
+                    return reject(err);
+                }
 
-            if (existingTaskConfigs) {
-                tasksJson['tasks'] = tasksJson['tasks'].concat(existingTaskConfigs);
-            }
+                resolve();
+            });
+        }
+        else {
+            // when tasks.json exists just update the tasks node
```

Comment:
>// when tasks.json exists just update the tasks node [](start = 12, length = 52)

Shouldn't we check if there are already tasks with the names we expect, and only add whatever tasks are missing?